test/fs/test_fs_mkdir_dotdot.c

```diff
@@ -0,0 +1,44 @@
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <string.h>
+#include <assert.h>
+#if defined(__EMSCRIPTEN__)
+#include <emscripten.h>
+#endif
+
+void makedir(const char *dir) {
+  int rtn = mkdir(dir, 0777);
+  assert(rtn == 0);
+}
+
+void changedir(const char *dir) {
+  int rtn = chdir(dir);
+  assert(rtn == 0);
+}
+
+void setup() {
+#if defined(__EMSCRIPTEN__) && defined(NODEFS)
+  makedir("working");
+  EM_ASM(FS.mount(NODEFS, { root: '.' }, 'working'));
+  changedir("working");
+#endif
+}
```

Comment:
Since this type of setup code seems common when testing nodefs, I wonder if we can remove it from the test file and put it some kind of `--pre-js` instead?